src/Illuminate/Foundation/Concerns/ResolvesDumpSource.php

```diff
@@ -34,16 +34,38 @@ public function resolveDumpSource()
         $relativeFile = $file;
 
         if (str_starts_with($file, $this->basePath)) {
-            $relativeFile = substr($file, strlen($this->basePath) + 1);
+            $relativeFile = $this->getAbsolutePath($file);
+        }
+
+        $viewPath = $this->getAbsolutePath(config('view.compiled'));
+
+        if (str_starts_with($relativeFile, $viewPath)) {
+            $fileArr = file($file);
+            $lastLine = end($fileArr);
+
+            $result = str_replace('<'.'?php /**PATH ', '', $lastLine);
+            $result = str_replace(' ENDPATH**/ ?>', '', $result);
+            $relativeFile = $this->getAbsolutePath($result);
```

Comment:
I would use a regex instead of `str_replace`.

Something like this will work `/\/\*\*PATH\s(.*)\sENDPATH/`.